Source/multi.cpp

```diff
@@ -609,35 +609,29 @@ void NetClose()
 	nthread_cleanup();
 	dthread_cleanup();
 	tmsg_cleanup();
-	multi_event_handler(0);
+	multi_event_handler(FALSE);
 	SNetLeaveGame(3);
 	msgcmd_cmd_cleanup();
 	if (gbMaxPlayers > 1)
 		Sleep(2000);
 }
 
-char multi_event_handler(int a1)
-{
-	int v1;                                                       // edi
-	void *(__stdcall * v2)(int, void(__stdcall *)(_SNETEVENT *)); // ebx
-	unsigned int v3;                                              // esi
-	int v4;                                                       // eax
-	char *v5;                                                     // eax
-
-	v1 = a1;
-	v2 = SNetRegisterEventHandler;
-	if (!a1)
-		v2 = SNetUnregisterEventHandler;
-	v3 = 0;
-	do {
-		v4 = (int)v2(event_types[v3], multi_handle_events);
-		if (!v4 && v1) {
-			v5 = TraceLastError();
-			app_fatal("SNetRegisterEventHandler:\n%s", v5);
+void multi_event_handler(BOOL add)
+{
+	DWORD i;
+	BOOL(__stdcall * fn)

```

Comment:
Would prefer to have these on the same line, but I guess `clang-format` split them?

```c
BOOL (__stdcall *fn)(int, void(__stdcall *)(_SNETEVENT *)); 
```

Reply:
If you want to go the extra mile, there's supposed to be a typdef according to the PSX:
`// TPDEF PTR FCN VOID SEVTHANDLER`
Something like:
```
typdef void (STORMAPI *SEVTHANDLER)(_SNETEVENT *);
// in the code
BOOL (STORMAPI *fn)(int, SEVTHANDLER);
```

Reply:
Sure, I like walking

Reply:
I take a space ship instead :P

Reply:
Does this look about right?
https://github.com/diasurgical/devilution/pull/1143